### DISCUSSION TITLE
Issues encountered while upgrading from v3.4.0 to v3.5.2

### DISCUSSION THREAD

**Discussion by dikshant-bansal** [facebook/docusaurus#10555] [General] [answered]
Hello,

I was trying to upgrade my existing docusaurus project from v3.4.0 to v3.5.2 when I encountered few issues.

1. When I migrated the packages to v3.5.2 getting errors related to '@theme/Unlisted' Component
Module not found: Error: Can't resolve '@theme/Unlisted' in '/Users/dikshantbansal/dev/developerdocs/src/theme/BlogPostPage'
<img width="1317" alt="image" src="https://github.com/user-attachments/assets/dfe78ad7-81fe-482e-80a8-545c60254338">

2. Updated imports on BlogPostPage and DocItem/Layout.
Replaced Unlisted Component with -
import ContentVisibility from '@theme/ContentVisibility'; 
<ContentVisibility metadata={metadata} />

But still getting error as
ERROR
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
<img width="1461" alt="image" src="https://github.com/user-attachments/assets/0a1ea8ca-a7dc-4004-a7e5-c0b3e599b00d">

Can someone guide here. Thanks.

**Comment by slorber** [comment]
Try running `docusaurus clear`

We can't help without a minimal repro, the problem could be anything

**Reply by dikshant-bansal** [reply]
Hey @slorber ,
I already tried the same. Unfortunately, didn't resolved the issue.

**Reply by slorber** [reply]
No repro, no help

**Reply by dikshant-bansal** [reply]
Apparently, I found that there are number of hooks and functions that have been moved from `@docusaurus/theme-common/internal` to `@docusaurus/plugin-content-docs/client `and  `@docusaurus/plugin-content-blog/client`.
Had to go in almost every swizzled component and update the imports.
This worked for me!

**Comment by dikshant-bansal** [comment, marked as answer]
I found that there are number of hooks and functions that have been moved from `@docusaurus/theme-common/internal` to `@docusaurus/plugin-content-docs/client `and  `@docusaurus/plugin-content-blog/client`.
Had to go in almost every swizzled component and update the imports.
This worked for me!
